src/System.Net.Sockets/src/System/Net/Sockets/BaseOverlappedAsyncResult.Unix.cs

```diff
@@ -23,7 +23,7 @@ public BaseOverlappedAsyncResult(Socket socket, Object asyncState, AsyncCallback
             if (NetEventSource.IsEnabled) NetEventSource.Info(this, socket);
         }
 
-        protected void CompletionCallback(int numBytes, SocketError errorCode)
+        internal void CompletionCallback(int numBytes, SocketError errorCode)
```

Comment:
Meaning... you think we should revert to what @luhenry had initially, with this protected, making the derived type partial, adding to the Unix-specific version of it a method that just delegates to the base, etc.?